### PR TITLE
Us040 login system

### PR DESCRIPTION
**Pull request by ryanbeatt**

### CODE REVIEW

**Review by cderry1 — APPROVED** [review]
Looks Good!!

File: app.ts [hunk 1]
```diff
@@ -1,19 +1,19 @@
...
+import crypto from 'crypto'
+
 const app: Express = express()
 
 const appViews = path.join(__dirname, '/views/')
 
 const nunjucksConfig = {
   autoescape: true,
-  noCahce: true,
+  noCache: true, // Fixed the typo in the configuration property name
```
Comment (ccurran01) [inline app.ts:15]: Can remove the comment, that's the benefit of Typescript, spot errors like that sooner

File: src/services/authService.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+const apiUrl: string = 'http://localhost:8080'
+
+export const login = async (login: Login): Promise<string> => {
+  try {
+    const response = await axios.post(apiUrl + '/api/login', login)
+    return response.data
+  } catch (error) {
+    if (axios.isAxiosError(error)) {
+      const axiosError = error as AxiosError
+      if ((axiosError.response != null) && axiosError.response.status === 401) {
```
Comment (ccurran01) [inline src/services/authService.ts:13]:
```suggestion
      if (axiosError.response?.status === 401) {
```

File: src/sinon.d.ts [hunk 3]
```diff
@@ -0,0 +1,4 @@
+declare module 'sinon' {
```
Comment (ccurran01) [inline src/sinon.d.ts:1]: What was this for?

**Review by ccurran01 — APPROVED** [review]

**Pull request merged**